sim/paladin/paladin.go

```diff
@@ -0,0 +1,175 @@
+package paladin
+
+import (
+	"time"
+
+	"github.com/wowsims/sod/sim/core"
+	"github.com/wowsims/sod/sim/core/proto"
+	"github.com/wowsims/sod/sim/core/stats"
+)
+
+const (
+	SealDuration                = time.Second * 30
+	SpellFlagSecondaryJudgement = core.SpellFlagAgentReserved1
+	SpellFlagPrimaryJudgement   = core.SpellFlagAgentReserved2
+)
+
+var TalentTreeSizes = [3]int{14, 15, 15}
+
+const (
+	SpellCode_PaladinNone = iota
+	SpellCode_PaladinHolyShock
+)
+
+type Paladin struct {
+	core.Character
+	Talents *proto.PaladinTalents
+
+	PaladinAura proto.PaladinAura
+
+	CurrentSeal           *core.Aura
+	CurrentSealExpiration time.Duration
+	PrimarySealSpell      *core.Spell
+
+	// Variables for max rank seal spells, used in APL Actions.
+	MaxRankRighteousness int
+	MaxRankCommand       int
+
+	// Active abilities and shared cooldowns that need externally manipulated.
+	CrusaderStrike    *core.Spell
+	DivineStorm       *core.Spell
+	Consecration      []*core.Spell
+	Exorcism          []*core.Spell
+	ExorcismCooldown  *core.Cooldown
+	HolyShock         []*core.Spell
+	HolyShockCooldown *core.Cooldown
+	Judgement         *core.Spell
+	DivineFavor       *core.Spell
+	// HolyShield            *core.Spell
+	// HammerOfWrath         *core.Spell
+	// HolyWrath             *core.Spell
+
+	// Seal spells and their associated auras
+	SealOfRighteousness []*core.Spell
+	SealOfCommand       []*core.Spell
+	SealOfMartyrdom     *core.Spell
+
+	SealOfRighteousnessAura []*core.Aura
+	SealOfCommandAura       []*core.Aura
+	SealOfMartyrdomAura     *core.Aura
+
+	// Auras from talents
+	DivineFavorAura *core.Aura
+	VengeanceAura   *core.Aura
+
+	// Placeholder for any auto-rotation with exo/HW.
+	DemonAndUndeadTargetCount int32
+}
+
+// Implemented by each Paladin spec.
+type PaladinAgent interface {
+	GetPaladin() *Paladin
+}
+
+func (paladin *Paladin) GetCharacter() *core.Character {
+	return &paladin.Character
+}
+
+func (paladin *Paladin) GetPaladin() *Paladin {
+	return paladin
+}
+
+func (paladin *Paladin) AddRaidBuffs(raidBuffs *proto.RaidBuffs) {
+	// Buffs are handled explicitly through APLs now
+}
+
+func (paladin *Paladin) AddPartyBuffs(_ *proto.PartyBuffs) {
+}
+
+func (paladin *Paladin) Initialize() {
+	// Update auto crit multipliers now that we have the targets.
+	paladin.AutoAttacks.MHConfig().CritMultiplier = paladin.MeleeCritMultiplier()
```

Comment:
I think this might be just a copy/paste leftover as I see you have no requirements to target in your CritMultiplier

Reply:
Correct. So this should just be
`	paladin.AutoAttacks.MHConfig().CritMultiplier = paladin.DefaultMeleeCritMultiplier()` 
and cutting out the defunct function call, right?

Reply:
yes